### PR TITLE
Setup2

### PR DESCRIPTION
**Pull request by z-fitch**
This is my version setup for reaching into CSV and setup for the gem files and spec harness. Let me know what you guys think

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -1,2 +1,24 @@
...
+              :away_goals,
+              :home_goals,
+              :venue_name,
+              :venue_link
+  def initialize(data)
+    @game_id = data[:game_id]
+    @season_id = data[:season]
+    @season_type = data[:type]
+    @game_date = data[:date_time]
+    @away_team_id = data[:away_team_id].to_i
```
Comment (awiley33) [inline lib/game.rb:17]: The conversion to integer here is good, we might want to do the same with all number values including the IDs.

File: lib/game_team.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+              :face_off_win_percentage, 
+              :giveaways, 
+              :takeaways
+
+  def initialize(data)
+    @game_id = data[:game_id]
+    @team_id = data[:team_id]
+    @home_or_away = data[:hoa]
+    @game_result = data[:result]
+    @reg_or_ot = data[:settled_in]
```
Comment (awiley33) [inline lib/game_team.rb:23]: I like the more specific instance var names here, it's more readable and I can understand what it's asking without having to go back to the CSV.

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+attr_reader :games, :teams, :game_teams, :seasons, :league
+
+def self.from_csv(locations)
+  StatTracker.new(locations)
+end
+
+def initialize(data)
+  @games = game_init(data)
+  @teams = teams_init(data)
+  @game_teams = game_teams_init(data)
```
Comment (awiley33) [inline lib/stat_tracker.rb:16]: I like the helper methods! Makes for a cleaner and more readable initialize method.

File: lib/season.rb [hunk 4]
```diff
@@ -1,2 +1,8 @@
 class Season
+  attr_reader :games
+  
+  def initialize(game)
+    @games = game
+  end
```
Comment (awiley33) [inline lib/season.rb:6]: Not sure if we will make it to the point of refactoring into these classes but we can always delete them later

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+def game_teams_init(data)
+  CSV.read(data[:game_teams], headers: true, header_converters: :symbol).map {|row| GameTeam.new(row) }
+end
+
+# GAME STATS
+
+  def highest_total_score
+    output = @games.max do |game1, game2|
+      # require 'pry'; binding.pry
```
Comment (awiley33) [inline lib/stat_tracker.rb:38]: This is one of my methods, just wanted to mention that I may look into refactoring this once I finish the other stats.

**Review by awiley33 — APPROVED** [review]
Overall looks good!! Please see comments.

File: lib/game.rb (hunk 1 above)
Comment (MWoodshop) [reply lib/game.rb:17]: These variables looks good, I think they are specific enough but not overly complicated. We might need to convert the other integers to_i if any of the others aren't already integers.

File: lib/stat_tracker.rb (hunk 3 above)
Comment (MWoodshop) [reply lib/stat_tracker.rb:16]: This is a good way to do this I think.

File: lib/game_team.rb (hunk 2 above)
Comment (MWoodshop) [reply lib/game_team.rb:23]: Agreed with these variables - we worked through these on a paired/meeting.

**Review by MWoodshop — APPROVED** [review]
Looks like a solid framework!

**Pull request merged**